File: example/src/test/java/com/example/util/crypto/CryptoUtilTest.java
package com.example.util.crypto;

import com.zipe.util.crypto.AesUtil;
import com.zipe.util.crypto.Base64Util;
import com.zipe.util.crypto.CryptoUtil;
import com.zipe.util.crypto.DESedeUtil;
import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.nio.charset.StandardCharsets;

@Slf4j
public class CryptoUtilTest {

    @Test
    public void aesTest() {
        // The secret key must have 16 bytes
        String secretKey = "1234567890123456";
        String oriContent = "Gary";
        CryptoUtil cryptoUtil = new CryptoUtil(new AesUtil(secretKey));
        String encryptMsg = cryptoUtil.encrypt(oriContent, StandardCharsets.UTF_8.name());
        log.info("AESEncrypt:{}", encryptMsg);

        String decryptMsg = cryptoUtil.decode(encryptMsg, StandardCharsets.UTF_8.name());
        log.info("AESDecrypt:{}", decryptMsg);

        Assertions.assertEquals(oriContent, decryptMsg);
    }

    @Test
    public void tripleTest() {
        // The secret key must have 48 bytes
        String secretKey = "098f6bcd4621d373cade4e832627b4f62017121611345734";
        String oriContent = "Gary";
        CryptoUtil cryptoUtil = new CryptoUtil(new DESedeUtil(secretKey));
        String encryptMsg = cryptoUtil.encrypt(oriContent, StandardCharsets.UTF_8.name());
        log.info("3DESEncrypt:{}", encryptMsg);

        String decryptMsg = cryptoUtil.decode(encryptMsg, StandardCharsets.UTF_8.name());
        log.info("3DESDecrypt:{}", decryptMsg);

        Assertions.assertEquals(oriContent, decryptMsg);
    }

    @Test
    public void base64Test() {
        String oriContent = "Gary";
        CryptoUtil cryptoUtil = new CryptoUtil(new Base64Util());
        String encryptMsg = cryptoUtil.encrypt(oriContent, StandardCharsets.UTF_8.name());
        log.info("Base64Encrypt:{}", encryptMsg);

        String decryptMsg = cryptoUtil.decode(encryptMsg, StandardCharsets.UTF_8.name());
        log.info("Base64Decrypt:{}", decryptMsg);

        Assertions.assertEquals(oriContent, decryptMsg);
    }
}

File: example-kotlin/src/test/kotlin/com/example/util/crypto/CryptoUtilTest.kt
package com.example.util.crypto

import com.zipe.util.crypto.AesUtil
import com.zipe.util.crypto.Base64Util
import com.zipe.util.crypto.CryptoUtil
import com.zipe.util.crypto.DESedeUtil
import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import org.slf4j.LoggerFactory
import java.nio.charset.StandardCharsets

/**
 * base-starter 加解密工具測試：分別以 AES、3DES、Base64 編解碼後，
 * 驗證解密結果與原始內容一致。
 */
class CryptoUtilTest : FunSpec({

    val log = LoggerFactory.getLogger(CryptoUtilTest::class.java)

    test("aesTest") {
        // The secret key must have 16 bytes
        val secretKey = "1234567890123456"
        val oriContent = "Gary"
        val cryptoUtil = CryptoUtil(AesUtil(secretKey))
        val encryptMsg = cryptoUtil.encrypt(oriContent, StandardCharsets.UTF_8.name())
        log.info("AESEncrypt:{}", encryptMsg)

        val decryptMsg = cryptoUtil.decode(encryptMsg, StandardCharsets.UTF_8.name())
        log.info("AESDecrypt:{}", decryptMsg)

        decryptMsg shouldBe oriContent
    }

    test("tripleTest") {
        // The secret key must have 48 bytes
        val secretKey = "098f6bcd4621d373cade4e832627b4f62017121611345734"
        val oriContent = "Gary"
        val cryptoUtil = CryptoUtil(DESedeUtil(secretKey))
        val encryptMsg = cryptoUtil.encrypt(oriContent, StandardCharsets.UTF_8.name())
        log.info("3DESEncrypt:{}", encryptMsg)

        val decryptMsg = cryptoUtil.decode(encryptMsg, StandardCharsets.UTF_8.name())
        log.info("3DESDecrypt:{}", decryptMsg)

        decryptMsg shouldBe oriContent
    }

    test("base64Test") {
        val oriContent = "Gary"
        val cryptoUtil = CryptoUtil(Base64Util())
        val encryptMsg = cryptoUtil.encrypt(oriContent, StandardCharsets.UTF_8.name())
        log.info("Base64Encrypt:{}", encryptMsg)

        val decryptMsg = cryptoUtil.decode(encryptMsg, StandardCharsets.UTF_8.name())
        log.info("Base64Decrypt:{}", decryptMsg)

        decryptMsg shouldBe oriContent
    }
})
